### PR TITLE
UrlDispatcher.add_route with partial coroutine handler

### PR DESCRIPTION
**Pull request by banthonywalker**
fix for problem when partialy applied coroutine is used as handler for UrlDispatcher.add_route

similar issue:
http://bugs.python.org/issue23519

### CODE REVIEW

File: aiohttp/web_urldispatcher.py [hunk 1]
```diff
@@ -93,7 +93,13 @@ def __init__(self, method, handler, *,
               issubclass(handler, AbstractView)):
             pass
         else:
-            handler = asyncio.coroutine(handler)
+            @asyncio.coroutine
+            def handler_wrapper(*args, **kwargs):
+                result = handler(*args, **kwargs)
+                if asyncio.iscoroutine(result):
```
Comment (asvetlov) [inline aiohttp/web_urldispatcher.py:99]: Please add a check for `asyncio.Future` also.

**Comment by asvetlov** [comment]
Tests needed.

File: aiohttp/web_urldispatcher.py (hunk 1 above)
Comment (ghost) [inline aiohttp/web_urldispatcher.py:99]: according to current documentation (for UrlDispatcher.add_route and Resource.add_route) handler should be callable which excludes asyncio.Future

**Comment by ghost** [comment]
test added

**Comment by asvetlov** [comment]
thanks

**Pull request merged**